# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dialog/nice_scroll.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./menubox.c.orig	2003-03-09 12:09:44.000000000 +0100
+++ ./menubox.c	2003-03-09 12:24:36.000000000 +0100
@@ -259,6 +259,8 @@
 	scrollamt = choice - (max_choice - 1);
 	choice = max_choice - 1;
     }
+    while ( (scrollamt+max_choice < item_no) &&
+	(choice > max_choice/2) ) { scrollamt++; choice--; }
 
     /* Print the menu */
     for (i = 0; i < max_choice; i++)
